src/app/shared/table/table.component.html

```diff
@@ -9,7 +9,7 @@
           <!-- Action Button -->
           <button mat-raised-button *ngIf="actionDef.type === 'button' && actionDef.id !== 'reset_filters'"
             (click)="actionTriggered(actionDef)" [color]="(actionDef.color ? actionDef.color : '')"
-            [disabled]="actionDef.disabled" [title]="(actionDef.name ? '' : actionDef.tooltip | translate)">
+    [disabled]="actionDef.disabled|| (dataSource.tableDef.rowSelection && actionDef.linkedToListSelection && dataSource.selectedRows === 0)" [title]="(actionDef.name ? '' : actionDef.tooltip | translate)">
```

Comment:
I don't think all the buttons are related to row selection (eg. export)
So this is why I think we need a new type, right?

Reply:
> I don't think all the buttons are related to row selection (eg. export)
> So this is why I think we need a new property as describe in the other comment, right?
> 
> > Enabling/Disabling a toolbar button according an item selection in the list should be done in a generic way by adding a new property in TableActionDef like 'linkedToListSelection: boolean' and use this prop in TableDataSource class to enable/disable automatically all these buttons.

I'v added the property linkedToListSelection as requested so we can distinguish them from those whom are not related to the row selection.